### PR TITLE
Cockpit 284

### PR DESCRIPTION
**Pull request by garrett**
Preview: https://garrett.github.io/cockpit-project.github.io/blog/cockpit-284.html

 - [x] [c-podman bodhi builds](https://bodhi.fedoraproject.org/updates/?packages=cockpit-podman)
 - [x] c-machines Fedora PRs land
 - [x] machines [bodhi URLs](https://bodhi.fedoraproject.org/updates/?packages=cockpit-machines)

### CODE REVIEW

**Comment by garrett** [comment]
I need to untag some PRs from last release still.

File: _posts/2023-01-25-cockpit-284.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+## Services: Show logs for user units
+
+The Services page now shows journal logs for user systemd services.
+
+![screenshot of show logs for user units]({{ site.baseurl }}/images/284-services-show-logs-for-user-units.png)
+
+*Many thanks to [Timothy Johnson](https://github.com/t1m0thyj) for this contribution!*
+
+## Storage: Set up a system to use NBDE
```
Comment (garrett) [inline _posts/2023-01-25-cockpit-284.md:25]:
```suggestion
## Storage: Set up NBDE
```

File: _posts/2023-01-25-cockpit-284.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+---
+title: Cockpit 284
+author: pitti
+date: '2023-01-25'
+tags: cockpit, machines
+slug: cockpit-284
+category: release
+summary: ''
```
Comment (garrett) [inline _posts/2023-01-25-cockpit-284.md:8]:
```suggestion
summary: 'User logs, NBDE, forceful snapshot reverting, and improved default container command'
```

File: _posts/2023-01-25-cockpit-284.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+tags: cockpit, machines
+slug: cockpit-284
+category: release
+summary: 'User logs, NBDE, forceful snapshot reverting, and improved default container command'
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
+We release regularly.
+
+Here are the release notes from Cockpit 284 and cockpit-machines 282:
```
Comment (garrett) [inline _posts/2023-01-25-cockpit-284.md:14]:
```suggestion
Here are the release notes from Cockpit 284, cockpit-machines 282, and cockpit-podman 284:
```

**Comment by martinpitt** [comment]
@garrett Other than the missing bodhi URLs (still working on the downstream releases), are you happy with the text now? Looks good enough to me.

File: _posts/2023-01-25-cockpit-284.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+The Services page now shows journal logs for user systemd services.
+
+![screenshot of show logs for user units]({{ site.baseurl }}/images/284-services-show-logs-for-user-units.png)
+
+*Many thanks to [Timothy Johnson](https://github.com/t1m0thyj) for this contribution!*
+
+## Storage: Set up NBDE
+
+It is possible (and unfortunately likely) that a system is misconfigured and can not mount filesystems during boot that rely on Clevis to unlock them. Cockpit can now fix this when adding a keyserver to a encrypted filesystem.
```
Comment (garrett) [inline _posts/2023-01-25-cockpit-284.md:27]:
```suggestion
A common misconfiguration of NBDE can prevent mounting filesystems that depend on Clevis. Cockpit now detects and fixes this issue while adding a keyserver to an encrypted filesystem.
```

File: _posts/2023-01-25-cockpit-284.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+![screenshot of set up a system to use nbde]({{ site.baseurl }}/images/284-storage-set-up-a-system-to-use-nbde.png)
+
+## Machines: Option to forcefully revert a snapshot
+
+There are situations where a snapshot revert involves extra risk and the user should either consider not reverting snapshot, or force it. The few most common cases are:
+- Snapshot that lacks full domain information for reverting configuration. In this case, the user should be sure that snapshot is compatible with the current configuration (if it is not, the domain will likely fail to run).
+- Reverting from a running domain to an active state where a new hypervisor has to be created rather than reusing the existing hypervisor.
+- Snapshot of inactive VM which also has a managed saved state. This requires forceful revert because the snapshot does not contain a memory state and will therefore not replace the existing memory state. This ends up switching a disk underneath a running system and can cause filesystem corruption or crashes due to swap content mismatches when run.
```
Comment (garrett) [inline _posts/2023-01-25-cockpit-284.md:36]: I think this is far too verbose. If this is really relevant and isn't addressed by the UI, then we should really try to do a follow-up. I don't think this should be documented in the release notes like this.

```suggestion
Cockpit can now revert virtual machine snapshots.
```

**Review by garrett — COMMENTED** [review]
A couple of suggestions.

File: _posts/2023-01-25-cockpit-284.md (hunk 5 above)
Comment (martinpitt) [reply _posts/2023-01-25-cockpit-284.md:36]: It was able to do that before, the new feature is that you can do it also in situations which the normal feature disallowed until now. I.e. UI for adding a `--force` flag.

Comment (martinpitt) [reply _posts/2023-01-25-cockpit-284.md:36]:
```suggestion
In some situations, reverting a snapshot involves extra risk and the user should either consider not reverting, or force it. The Machines page can now do this, and explains the reason.
```

Comment (garrett) [reply _posts/2023-01-25-cockpit-284.md:36]: OK, this merges the two suggestions, with some revisions.

```suggestion
Cockpit can now forcefully revert virtual machine snapshots.

Reverting a snapshot may involve risk. The revert dialog will explain these corner cases with a message, to help determine if forcing a revert should be preferred or to consider not reverting a snapshot.
```

**Review by martinpitt — APPROVED** [review]
All bodhi URLs added. Go!

**Pull request merged**